projects/frontend/data-pipelines/gui/projects/ui/src/app/app.component.spec.ts

```diff
@@ -63,24 +53,42 @@ describe('AppComponent', () => {
         TestBed.configureTestingModule({
             schemas: [NO_ERRORS_SCHEMA],
             declarations: [AppComponent],
-            imports: [],
             providers: [
+                { provide: AppConfigService, useValue: appConfigServiceStub },
                 UrlHelperService,
+                DynamicComponentsService,
+                ConfirmationService,
+                UrlOpenerService,
                 { provide: OAuthService, useValue: oAuthServiceStub },
                 { provide: NavigationService, useValue: navigationServiceStub },
-                { provide: RouterService, useValue: routerServiceStub },
-                { provide: ViewContainerRef, useValue: viewContainerRefStub },
-                { provide: DynamicComponentsService, useValue: dynamicComponentsServiceStub },
-                { provide: ConfirmationService, useValue: confirmationServiceStub },
-                { provide: UrlOpenerService, useValue: urlOpenerServiceStub },
-                { provide: AppConfigService, useValue: appConfigServiceStub }
+                { provide: RouterService, useValue: routerServiceStub }
             ]
         });
     });
 
-    it('should create the app', () => {
+    it('should create the app with no components ignored', () => {
+        appConfigServiceStub.getConfig.and.returnValue({
+            auth: {},
+            ignoreRoutes: [],
+            ignoreComponents: []
+        });
+
+        const fixture = TestBed.createComponent(AppComponent);
+        const app = fixture.componentInstance;
+        expect(app).toBeTruthy();
+        expect(app.explorePageVisible).toBeTrue();
+    });
+
+    it('should create the app with explore page ignored', () => {
+        appConfigServiceStub.getConfig.and.returnValue({
+            auth: {},
+            ignoreRoutes: [],
```

Comment:
May you elaborate the purpose of ignoreRoutes?

Reply:
if you ignore just the explore page component in the html, it won't be visible. However, if you navigate to `/explore/data-jobs` in the browser, you will still get the page. We have to ignore the route to prevent this from happening.

It works locally, but I couldn't figure out a good way to test it in the spec. Any ideas are welcome. 😕

Reply:
Added the test in cypress. It passes locally. Let's see if it will pass in CI.